crates/shared/src/event_handling.rs

```diff
@@ -286,19 +290,26 @@ where
 
     async fn update_events_from_latest_blocks(
         &mut self,
-        blocks: &[BlockNumberHash],
+        latest_blocks: &[BlockNumberHash],
         is_reorg: bool,
     ) -> Result<()> {
-        let (blocks, events) = self.past_events_by_block_hashes(blocks).await;
-        track_block_range(&format!("range_{}", blocks.len()));
-        tracing::debug!(
-            "final blocks for updating: {:?} - {:?}",
-            blocks.first(),
-            blocks.last()
+        anyhow::ensure!(
```

Comment:
Should this be a:
```suggestion
        debug_assert!(
```

Since it should never happen as it is only called if `!event_range.latest_blocks.is_empty()`?

I think its fine to panic in case of logic errors that aren't dependent on "untrusted" input.

Reply:
Resolved.